src/kv/kv_types.h

```diff
@@ -417,6 +422,23 @@ namespace kv
         return snapshots;
       }
 
+      std::vector<uint8_t>& get_buffer()
```

Comment:
This is introducing quite a lot of implementation in `kv_types.h`. Could we create a new `kv/snapshot.h` file instead?